### PR TITLE
Like, WatchLater and History

### LINKED ISSUE #11: Like, WatchLater and History Functionality
- User should be able to like a video and unlike a video
- Liked video's should appear in liked route `/liked`
- Should be able to add videos to Watch Later
- Video should be added to history once played
- Clear history

### PR DESCRIPTION
**Pull request by ttahm3d**
- Added Like, WatchLater and History pages
- Created functions in context and setup reducer
- Fully responsive for all screen sizes.

Fix for issue #11

### CODE REVIEW

**Review by Hamdrive — COMMENTED** [review]
Great work implementing the logic for liked videos and history
Neat use of context and reducer to handle state updates
Lots to learn from your code, great logic and practices followed

**Review by omgaikwad — APPROVED** [review]
Good file structure, web app is mobile responsive, clean code, use of context and reducer is done, great job!

File: src/context/videos/Reducer.jsx [hunk 1]
```diff
@@ -2,6 +2,16 @@ const videosReducer = (state, { type, payload }) => {
   switch (type) {
     case "FILTER_BY_CATEGORY":
       return { ...state, category: payload };
+    case "ADD_TO_LIKED":
+    case "REMOVE_FROM_LIKED":
```
Comment (coldpigli) [inline src/context/videos/Reducer.jsx:6]: Nice move here. Not using break or default for the action types that perform the same thing. This way the developer knows what's happening without repeating the code for same thing.

File: src/pages/history/History.jsx [hunk 2]
```diff
@@ -1,10 +1,35 @@
+import { Button, Empty } from "../../components";
+import { useVideos } from "../../context/videos/Context";
 import { useDocumentTitle } from "../../hooks";
```
Comment (coldpigli) [inline src/pages/history/History.jsx:3]: Liked the use of useDocumentTitle custom hook, to get document titles dynamically

**Review by coldpigli — APPROVED** [review]
Learnt a few things from your code.
Have mentioned what I liked.
Will try to implement it in the next iteration

**Pull request merged**